addons/gamification/models/gamification_challenge.py

```diff
@@ -61,233 +62,218 @@ class gamification_challenge(osv.Model):
+
+    invited_user_ids = fields.Many2many('res.users', 'gamification_invited_user_ids_rel', string="Suggest to users")
+
+    line_ids = fields.One2many(
+        'gamification.challenge.line', 'challenge_id',
+        string='Lines', help="List of goals that will be set", required=True, copy=True)
+
+    reward_id = fields.Many2one('gamification.badge', string="For Every Succeding User")
+    reward_first_id = fields.Many2one('gamification.badge', string="For 1st user")
+    reward_second_id = fields.Many2one('gamification.badge', string="For 2nd user")
+    reward_third_id = fields.Many2one('gamification.badge', string="For 3rd user")
+    reward_failure = fields.Boolean('Reward Bests if not Succeeded?')
+    reward_realtime = fields.Boolean(
+        'Reward as soon as every goal is reached', default=True,
+        help="With this option enabled, a user can receive a badge only once. The top 3 badges are still rewarded only at the end of the challenge.")
+
+    visibility_mode = fields.Selection(
+        [
+            ('personal', 'Individual Goals'),
+            ('ranking', 'Leader Board (Group Ranking)'),
+        ], default='personal', string="Display Mode", required=True)
+
+    report_message_frequency = fields.Selection(
+        [
+            ('never', 'Never'),
+            ('onchange', 'On change'),
+            ('daily', 'Daily'),
+            ('weekly', 'Weekly'),
+            ('monthly', 'Monthly'),
+            ('yearly', 'Yearly')
+        ], default='never', string="Report Frequency", required=True)
+    report_message_group_id = fields.Many2one(
+        'mail.channel', string='Send a copy to',
+        help='Group that will receive a copy of the report in addition to the user')
+    report_template_id = fields.Many2one('mail.template', string="Report Template", required=True, default=_get_report_template)
+    remind_update_delay = fields.Integer(
+        string='Non-updated manual goals will be reminded after',
+        help="Never reminded if no value or zero is specified.")
+    last_report_date = fields.Date(default=fields.Date.today())
+    next_report_date = fields.Date(compute="_compute_next_report_date", store=True)
+
+    category = fields.Selection(
+        '_get_categories', default='hr',
+        string="Appears in", help="Define the visibility of the challenge through menus", required=True)
+
+    @api.depends('last_report_date', 'report_message_frequency')
+    def _compute_next_report_date(self):
         """Return the next report date based on the last report date and report
         period.
 
         :return: a string in DEFAULT_SERVER_DATE_FORMAT representing the date"""
-        res = {}
-        for challenge in self.browse(cr, uid, ids, context=context):
-            last = datetime.strptime(challenge.last_report_date, DF).date()
+        for challenge in self:
+            last = fields.Datetime.from_string(challenge.last_report_date)
```

Comment:
last_report_date is date field not datetime. 
so Use: 
last = fields.Date.from_string(challenge.last_report_date)